svcdiscovery/plugin/svcdiscovery/setup.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2020 SUSE
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package svcdiscovery
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/caddyserver/caddy"
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/plugin"
+	clog "github.com/coredns/coredns/plugin/pkg/log"
+)
+
+const pluginName = "svcdiscovery"
+const sdcEndpointBase = "/v1/registration/"
+
+// Register the plugin.
+func init() {
+	plugin.Register(pluginName, func(c *caddy.Controller) error {
+		// Ignore svcdiscovery token.
+		c.Next()
+
+		var sdcClient *SDCClient
+		var tlsCAPath string
+		var tlsClientCertPath string
+		var tlsClientKeyPath string
+		var sdcHost string
+		var sdcPort string
+		var ttl uint32
+		for c.NextBlock() {
+			switch c.Val() {
+			case "tls_ca_path":
```

Comment:
Most of these are a plain string; they look very repetitive, and I wonder if it would be possible to reduce some of the repetition without making the code too much harder to read.  Also, do we need to check that `sdcPort` is a proper number? Or that we're not getting values that make no sense (since we don't have a `default` block in the `switch`).

Reply:
1. +100 on missing `default` statement. I'm usually the one to complain when it is missing.
2. Repetition is not inherently bad. I don't want to spend time trying to simplify this since it's already very simple, just repetitive.
3. I can certainly do validation on the `sdcPort`.

Reply:
Added validation to `sdc_port` with b2841b9.

Reply:
Added missing `default` case with 5daee74.